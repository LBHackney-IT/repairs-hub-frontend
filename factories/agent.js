import { Factory } from 'fishery'

const agentUserFactory = Factory.define(() => ({
  name: 'An Agent',
  email: 'an.agent@hackney.gov.uk',
  roles: ['agent'],
  hasRole: true,
  hasAgentPermissions: true,
  hasContractorPermissions: false,
  hasContractManagerPermissions: false,
  hasAuthorisationManagerPermissions: false,
  hasOperativePermissions: false,
  hasAnyPermissions: true,
}))

export const agent = agentUserFactory.build()
