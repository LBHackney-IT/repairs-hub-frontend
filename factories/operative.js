import { Factory } from 'fishery'

export const operativeUserFactory = Factory.define(() => ({
  name: 'An Agent',
  email: 'an.agent@hackney.gov.uk',
  roles: ['operative'],
  hasRole: true,
  hasAgentPermissions: false,
  hasContractorPermissions: false,
  hasContractManagerPermissions: false,
  hasAuthorisationManagerPermissions: false,
  hasOperativePermissions: true,
  hasAnyPermissions: true,
}))

export const operative = operativeUserFactory.build()
