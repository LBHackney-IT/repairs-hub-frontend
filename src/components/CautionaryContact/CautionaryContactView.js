import BackButton from '../Layout/BackButton'
import CautionaryContactTable from './CautionaryContactTable'

const CautionaryContactView = (query) => {
  return (
    <>
      <BackButton />
      <h3 className="lbh-heading-h3">Cautionary contact</h3>
      <CautionaryContactTable codes={query ? query.query.cautContact : false} />
    </>
  )
}

export default CautionaryContactView
