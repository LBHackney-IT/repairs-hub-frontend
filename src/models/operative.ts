export type Operative = {
  jobPercentage: number
  id: number
  payrollNumber: string
  name: string
  trades: string[]
  isOnejobatatime: boolean
}
