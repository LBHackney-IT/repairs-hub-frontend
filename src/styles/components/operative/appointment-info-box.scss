.appointment-info-box {
  border-top: 8px solid repairs-hub-colour('dark-green');
  border-bottom: 2px solid repairs-hub-colour('dark-green');
  border-right: 2px solid repairs-hub-colour('dark-green');
  border-left: 2px solid repairs-hub-colour('dark-green');
  position: relative;
  padding: 15px;
  overflow-y: auto;
  min-height: 0;
  width: 100%;
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;

  .status-text {
    margin-left: auto;
    margin-right: 20px;
    margin-top: 0;
    margin-bottom: 0;
  }
}
