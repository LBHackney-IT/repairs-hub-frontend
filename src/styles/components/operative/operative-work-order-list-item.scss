.operative-work-order-list-item {
  padding: 15px;
  box-sizing: border-box;
  display: flex;
  justify-content: space-between;
  margin-top: 15px;

  &--active {
    border-top: 8px solid repairs-hub-colour('dark-green');
    border-bottom: 2px solid repairs-hub-colour('dark-green');
    border-right: 2px solid repairs-hub-colour('dark-green');
    border-left: 2px solid repairs-hub-colour('dark-green');
  }

  &--inactive {
    background-color: repairs-hub-colour('light-grey');
  }

  .lbh-tag {
    font-weight: 600;
    vertical-align: top;
  }

  .lbh-tag--status-closed,
  .lbh-tag--status-completed {
    background-color: repairs-hub-colour('dark-grey');
  }

  .appointment-time {
    font-size: 24px;
  }

  .appointment-details {
    p {
      font-size: 18px;
    }
  }
}
