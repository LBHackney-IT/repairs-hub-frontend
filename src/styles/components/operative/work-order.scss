.operative-work-order {
  .person-alert {
    &--icon {
      background: repairs-hub-colour('dark-red');
      border: 3px solid repairs-hub-colour('dark-red');
    }

    &--text {
      color: repairs-hub-colour('dark-red');
    }
  }

  .align-grid-column {
    margin-top: 0rem;
  }

  .priority-text {
    text-align: right;
    font-size: 20px;

    .lbh-heading-h3 {
      font-weight: normal;
    }
  }

  .property-name {
    font-weight: 600;
  }
}

.truncate-description {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  word-break: var(--word-break, 'none');
  overflow: hidden;
}
.truncate-line-3 {
  -webkit-line-clamp: var(--line-clamp, 3);
}

.truncate-line-5 {
  -webkit-line-clamp: var(--line-clamp, 5);
}

.update-task-quantity {
  width: 50%;
}
