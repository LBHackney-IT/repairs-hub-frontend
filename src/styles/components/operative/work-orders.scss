.operative-work-orders {
  .govuk-grid-column-full {
    @include govuk-media-query($until: tablet) {
      padding: 0;
    }

    h1 {
      @include govuk-media-query($until: tablet) {
        padding: 0 15px;
      }
    }
  }
}
