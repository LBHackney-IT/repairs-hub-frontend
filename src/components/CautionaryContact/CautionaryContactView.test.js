import { render } from '@testing-library/react'
import UserContext from '../UserContext'
import CautionaryContactView from './CautionaryContactView'
import { operative } from 'factories/operative'
describe('CautionaryContactView component', () => {
  const query = {
    cautContact: ['VA', 'DAT', 'CIT'],
  }
  describe('when logged in as an operative and has NO queries', () => {
    it('should render properly: with NO highlithed codes', () => {
      const { asFragment } = render(
        <UserContext.Provider value={{ user: operative }}>
          <CautionaryContactView />
        </UserContext.Provider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })

  describe('when logged in as an operative and has queries', () => {
    it('should render properly: with highlithed codes', () => {
      const { asFragment } = render(
        <UserContext.Provider value={{ user: operative }}>
          <CautionaryContactView query={query} />
        </UserContext.Provider>
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
