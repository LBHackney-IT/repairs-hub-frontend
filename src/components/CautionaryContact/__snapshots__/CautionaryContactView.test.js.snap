// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CautionaryContactView component when logged in as an operative and has NO queries should render properly: with NO highlithed codes 1`] = `
<DocumentFragment>
  <a
    class="govuk-back-link lbh-back-link govuk-!-display-none-print"
    role="button"
  >
    Back
  </a>
  <h3
    class="lbh-heading-h3"
  >
    Cautionary contact
  </h3>
  <table
    class="govuk-table lbh-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row lbh-body"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Code
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
        data-row-id="0"
      >
        <td
          class="govuk-table__cell"
        >
          DAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="1"
      >
        <td
          class="govuk-table__cell"
        >
          CFT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="2"
      >
        <td
          class="govuk-table__cell"
        >
          CIT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="3"
      >
        <td
          class="govuk-table__cell"
        >
          CVT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="4"
      >
        <td
          class="govuk-table__cell"
        >
          OIT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="5"
      >
        <td
          class="govuk-table__cell"
        >
          PAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="6"
      >
        <td
          class="govuk-table__cell"
        >
          SAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="7"
      >
        <td
          class="govuk-table__cell"
        >
          UET
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="8"
      >
        <td
          class="govuk-table__cell"
        >
          VAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Verbal Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="9"
      >
        <td
          class="govuk-table__cell"
        >
          COR
        </td>
        <td
          class="govuk-table__cell"
        >
          ASBO or injunction obtained
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="10"
      >
        <td
          class="govuk-table__cell"
        >
          DA
        </td>
        <td
          class="govuk-table__cell"
        >
          Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="11"
      >
        <td
          class="govuk-table__cell"
        >
          CX
        </td>
        <td
          class="govuk-table__cell"
        >
          Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="12"
      >
        <td
          class="govuk-table__cell"
        >
          DV
        </td>
        <td
          class="govuk-table__cell"
        >
          Domestic Violence
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="13"
      >
        <td
          class="govuk-table__cell"
        >
          CF
        </td>
        <td
          class="govuk-table__cell"
        >
          No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="14"
      >
        <td
          class="govuk-table__cell"
        >
          CI
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="15"
      >
        <td
          class="govuk-table__cell"
        >
          CV
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="16"
      >
        <td
          class="govuk-table__cell"
        >
          OI
        </td>
        <td
          class="govuk-table__cell"
        >
          Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="17"
      >
        <td
          class="govuk-table__cell"
        >
          PA
        </td>
        <td
          class="govuk-table__cell"
        >
          Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="18"
      >
        <td
          class="govuk-table__cell"
        >
          SA
        </td>
        <td
          class="govuk-table__cell"
        >
          Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="19"
      >
        <td
          class="govuk-table__cell"
        >
          SPR
        </td>
        <td
          class="govuk-table__cell"
        >
          Specific Requirements
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="20"
      >
        <td
          class="govuk-table__cell"
        >
          CXT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="21"
      >
        <td
          class="govuk-table__cell"
        >
          TRA
        </td>
        <td
          class="govuk-table__cell"
        >
          Translation Required
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="22"
      >
        <td
          class="govuk-table__cell"
        >
          UE
        </td>
        <td
          class="govuk-table__cell"
        >
          Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="23"
      >
        <td
          class="govuk-table__cell"
        >
          VA
        </td>
        <td
          class="govuk-table__cell"
        >
          Verbal Abuse or Threat of
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`CautionaryContactView component when logged in as an operative and has queries should render properly: with highlithed codes 1`] = `
<DocumentFragment>
  <a
    class="govuk-back-link lbh-back-link govuk-!-display-none-print"
    role="button"
  >
    Back
  </a>
  <h3
    class="lbh-heading-h3"
  >
    Cautionary contact
  </h3>
  <table
    class="govuk-table lbh-table"
  >
    <thead
      class="govuk-table__head"
    >
      <tr
        class="govuk-table__row lbh-body"
      >
        <th
          class="govuk-table__header"
          scope="col"
        >
          Code
        </th>
        <th
          class="govuk-table__header"
          scope="col"
        >
          Description
        </th>
      </tr>
    </thead>
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row"
        data-row-id="0"
      >
        <td
          class="govuk-table__cell highlight-for-existing-code"
        >
          DAT
        </td>
        <td
          class="govuk-table__cell highlight-for-existing-code"
        >
          [Temporary] Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="1"
      >
        <td
          class="govuk-table__cell"
        >
          CFT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="2"
      >
        <td
          class="govuk-table__cell highlight-for-existing-code"
        >
          CIT
        </td>
        <td
          class="govuk-table__cell highlight-for-existing-code"
        >
          [Temporary] No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="3"
      >
        <td
          class="govuk-table__cell"
        >
          CVT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="4"
      >
        <td
          class="govuk-table__cell"
        >
          OIT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="5"
      >
        <td
          class="govuk-table__cell"
        >
          PAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="6"
      >
        <td
          class="govuk-table__cell"
        >
          SAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="7"
      >
        <td
          class="govuk-table__cell"
        >
          UET
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="8"
      >
        <td
          class="govuk-table__cell"
        >
          VAT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Verbal Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="9"
      >
        <td
          class="govuk-table__cell"
        >
          COR
        </td>
        <td
          class="govuk-table__cell"
        >
          ASBO or injunction obtained
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="10"
      >
        <td
          class="govuk-table__cell"
        >
          DA
        </td>
        <td
          class="govuk-table__cell"
        >
          Dangerous Animals
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="11"
      >
        <td
          class="govuk-table__cell"
        >
          CX
        </td>
        <td
          class="govuk-table__cell"
        >
          Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="12"
      >
        <td
          class="govuk-table__cell"
        >
          DV
        </td>
        <td
          class="govuk-table__cell"
        >
          Domestic Violence
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="13"
      >
        <td
          class="govuk-table__cell"
        >
          CF
        </td>
        <td
          class="govuk-table__cell"
        >
          No Female Staff to Visit
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="14"
      >
        <td
          class="govuk-table__cell"
        >
          CI
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Interviews
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="15"
      >
        <td
          class="govuk-table__cell"
        >
          CV
        </td>
        <td
          class="govuk-table__cell"
        >
          No Lone Visits
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="16"
      >
        <td
          class="govuk-table__cell"
        >
          OI
        </td>
        <td
          class="govuk-table__cell"
        >
          Other Type of Incident
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="17"
      >
        <td
          class="govuk-table__cell"
        >
          PA
        </td>
        <td
          class="govuk-table__cell"
        >
          Physical Abuse or Threat of
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="18"
      >
        <td
          class="govuk-table__cell"
        >
          SA
        </td>
        <td
          class="govuk-table__cell"
        >
          Seek Advice
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="19"
      >
        <td
          class="govuk-table__cell"
        >
          SPR
        </td>
        <td
          class="govuk-table__cell"
        >
          Specific Requirements
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="20"
      >
        <td
          class="govuk-table__cell"
        >
          CXT
        </td>
        <td
          class="govuk-table__cell"
        >
          [Temporary] Do Not Attend - Refer to Authorising Officer
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="21"
      >
        <td
          class="govuk-table__cell"
        >
          TRA
        </td>
        <td
          class="govuk-table__cell"
        >
          Translation Required
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="22"
      >
        <td
          class="govuk-table__cell"
        >
          UE
        </td>
        <td
          class="govuk-table__cell"
        >
          Unclean / Unsafe Environment
        </td>
      </tr>
      <tr
        class="govuk-table__row"
        data-row-id="23"
      >
        <td
          class="govuk-table__cell highlight-for-existing-code"
        >
          VA
        </td>
        <td
          class="govuk-table__cell highlight-for-existing-code"
        >
          Verbal Abuse or Threat of
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
